src/kagglehub/http_resolver.py

```diff
@@ -330,6 +342,9 @@ def _build_list_model_instance_version_files_url_path(h: ModelHandle) -> str:
 def _build_get_dataset_url_path(h: DatasetHandle) -> str:
     return f"datasets/view/{h.owner}/{h.dataset}"
 
+def _build_get_notebook_url_path(h: NotebookHandle) -> str:
+    return f"kernels/output/download/{h.owner}/{h.notebook}?version_number={h.version}"
```

Comment:
nit: Ack that you're copying existing datasets code, but this is slightly fragile in that it's silently assuming the optional field `h.version` is set, which is true for now but could lead to tricky bugs later.  We could put a check + `raise` here (and for datasets) to make that assumption explicit.  Or make the code robust to handle unset `h.version`.

Reply:
It is set whenever it checks what the current version is. However, I do understand that it could be trickier down the line